src/OrchardCore/Orchard.Users.Abstractions/IUser.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace Orchard.Users
+{
+    /// <summary>
+    /// Marker interface for ASP.NET Core Identity services.
+    /// </summary>
+    public interface IUser
+    {
```

Comment:
Should we have at least a few fundamental properties like a user ID?

Reply:
Usually, you will use services to get User by id like 
UserManager.FindByIdAsync(string userId);
By default asp.net identity use Id type of string. Or you may query concrete type from database like _session.Query<Orchard.Users.Models.User, UserIndex>();
We may add string or Integer as Id but it may not work for some Apps. I also thought about generics like IUser<TKey>, but it makes everything complicated, and just leave it without Id, let application decide what do they want. 
Asp.net identity for .net 4.X use strong contract IUser<TKey> it was hard to maintain and they decided to get rid of from it in .net identity for .net core.

Reply:
I'm fine with this pattern (that I also use in OpenIddict), but we should be consistent: `IRole` that is also added as part of this PR has a `RoleName` property (https://github.com/OrchardCMS/Orchard2/pull/810/files#diff-d306f670b897d68ba6af8ad0b8c35042R5) while the `IUser` contract is empty.

If we go with the empty interfaces approach, we may also consider removing them and using `object` or `dynamic` everywhere :sweat_smile:

Reply:
IUser is like a marker interface, we can register UserManager<object> but I think with the marker interface better UserManager<IUser>.

Reply:
We probably need to expand the contract IUser.

Reply:
All users must have a basic

```c#
public interface IUser
{
    public int Id { get; set; }
    public string UserName { get; set; }
    public string NormalizedUserName { get; set; }
    public string Email { get; set; }
    public string NormalizedEmail { get; set; }
    public string PasswordHash { get; set; }
    public string SecurityStamp { get; set; }
    public bool EmailConfirmed { get; set; }
    public List<string> RoleNames { get; set; }
}
```

Reply:
some Apps do not need NormalizedUserName, NormalizedEmail and just use Email instead of UserName.
PasswordHash  and SecurityStamp  I guess those are App Domain boundary specific too.

Reply:
I'm not sure about IUser.Id type too, I have an experience integrating Orchard1 membership system with remote store, they used UserId like a string.

Reply:
I would say `RoleName` and `UserName`, with just a `getter`. Because all properties are set using the services, and we don't need anything else as a minimum contract.

Reply:
It is necessary to think

Reply:
I guess for now we can just leave UserName.
RoleNames can be set/get via services, Authorization filter attributes use services to check if user belongs to a particular role. +If I decided to override IUserStore by using relation db each time I need a user I have to make JOINS in order to retrieve roles (even if I do not use them).
If it won't enough we may create another PR to fix it.